packages/plugin-edgedb/src/global-types.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable @typescript-eslint/no-unud-vars */
+import {
+  FieldNullability,
+  InputFieldMap,
+  InterfaceParam,
+  SchemaTypes,
+  TypeParam,
+} from '@pothos/core';
+import {
+  EdgeDBDriver,
+  EdgeDBModelShape,
+  EdgeDBObjectFieldOptions,
+  EdgeDBObjectTypeOptions,
+  EdgeDBQueryBuilder,
+  EdgeDBTypeKeys,
+} from './types';
+import type { EdgeDBPlugin } from '.';
+import { EdgeDBObjectFieldBuilder as InternalEdgeDBObjectFieldBuilder } from './edgedb-field-builder';
+import { EdgeDBObjectRef } from './object-ref';
+
+declare global {
+  export namespace PothosSchemaTypes {
+    export interface Plugins<Types extends SchemaTypes> {
+      edgedb: EdgeDBPlugin<Types>;
+    }
+
+    export interface UserSchemaTypes {
+      EdgeDBTypes: { default: { [key: string]: unknown } };
+    }
+
+    export interface ExtendDefaultTypes<PartialTypes extends Partial<UserSchemaTypes>> {
+      EdgeDBTypes: PartialTypes['EdgeDBTypes'] & {};
+    }
+
+    export interface PothosKindToGraphQLType {
+      EdgeDBObject: 'Object';
+    }
+
+    export interface SchemaBuilderOptions<Types extends SchemaTypes> {
+      edgeDB: {
+        qb: EdgeDBQueryBuilder;
+        client: EdgeDBDriver;
+        extensions?: {};
+        exposeDescriptions?:
+          | boolean
+          | {
+              types?: boolean;
+              fields?: boolean;
+            };
+      };
+    }
+
+    export interface BuildSchemaOptions<Types extends SchemaTypes> {
+      customBuildTimeOptions?: boolean;
+    }
+
+    export interface ObjectTypeOptions<Types extends SchemaTypes, Shape> {
+      optionOnObject?: boolean;
+    }
+
+    export interface FieldOptionsByKind<
+      Types extends SchemaTypes,
+      ParentShape,
+      Type extends TypeParam<Types>,
+      Nullable extends FieldNullability<Type>,
+      Args extends InputFieldMap,
+      ResolveShape,
+      ResolveReturnShape,
+    > {
+      EdgeDBObject: EdgeDBObjectFieldOptions<
+        Types,
+        ParentShape,
+        Type,
+        Nullable,
+        Args,
+        ResolveShape,
+        ResolveReturnShape
+      >;
+    }
+
+    export interface SchemaBuilder<Types extends SchemaTypes> {
+      edgeDBObject: <
+        Name extends EdgeDBTypeKeys<Types>,
+        Interfaces extends InterfaceParam<Types>[],
+        Model extends EdgeDBModelShape<Types, Name>,
+      >(
+        name: Name,
+        options: EdgeDBObjectTypeOptions<Types, Interfaces, Model>,
+      ) => EdgeDBObjectRef<Types, Model>;
+    }
+
+    export interface EdgeDBObjectFieldBuilder<
```

Comment:
I think I noticed that there is something broken about this in the Prisma plugin.  I tried referencing this global type to create a function that accepts a PrismaFieldBuilder, but it wasn't actually compatible with the PrismaFieldBuilder type.  Not sure if this will have the same issue, but something that might be worth checking on.  Not sure if you need this at all if you are not supporting extending EdgeDBObjectFieldBuilder from other plugins initially.